WordPress/src/main/java/org/wordpress/android/ui/posts/PostListDialogHelper.kt

```diff
@@ -18,7 +22,8 @@ private const val CONFIRM_ON_AUTOSAVE_REVISION_DIALOG_TAG = "CONFIRM_ON_AUTOSAVE
  */
 class PostListDialogHelper(
     private val showDialog: (DialogHolder) -> Unit,
-    private val checkNetworkConnection: () -> Boolean
+    private val checkNetworkConnection: () -> Boolean,
+    private val analyticsTracker: AnalyticsTrackerWrapper
```

Comment:
I am not a big fan of passing the `AnalyticsTrackerWrapper` to the `PostListDialogHelper` and I'd have personally preferred to move the positive dialog actions to their respective methods and handling the show dialog tracking in a different way, but I can't justify requesting such a change after the fact as I don't think there would be significant enough benefit to it. So, I am just leaving a note with my thoughts, but you should basically ignore this :)

Just for the sake of sharing, I was thinking of adding a `PostListDialogType` `enum` with tag and tracking info attached to it and passing that back to the main view model in `showDialog`, but yeah, not really worth it and might even end up being confusing 🤷‍♂

Reply:
Thanks for sharing! ;)

I think the solution you are proposing would be nice, but at the same time I agree with you that it's probably not worth the changes. I also feel a bit that both logging and tracking aren't real dependencies -> I don't mind, that much, if they are used even in classes where they "don't belong".